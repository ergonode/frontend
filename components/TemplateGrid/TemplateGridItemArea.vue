/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
<template>
    <div
        :item-id="item.id"
        :style="gridItemStyles"
        class="grid__item-area">
        <slot />
        <slot name="connection" />
    </div>
</template>
<script>
export default {
    name: 'TemplateGridItemArea',
    props: {
        item: {
            type: Object,
            required: true,
        },
        columns: {
            type: Number,
            required: true,
        },
        gridGap: {
            type: Number,
            default: 0,
        },
    },
    computed: {
        gridItemStyles() {
            const { column } = this.item;
            return {
                gridColumn: `${column + 1} / ${this.columns + 1}`,
                padding: `${this.gridGap}px`,
                gridTemplateColumns: `repeat(${this.columns - column}, 1fr)`,
                gridGap: `${this.gridGap}px`,
            };
        },
    },
};
</script>

<style lang="scss" scoped>
    .grid__item-area {
        position: relative;
        display: grid;
        justify-content: flex-start;
        align-items: center;

        .item-area__line {
            position: absolute;
            z-index: $Z_INDEX_NEGATIVE;
            display: flex;
            width: 100%;
            grid-column: 1 / 1;
            border-left: $BORDER_2_GREEN;
            border-bottom: $BORDER_2_GREEN;
            transition: width 0.3s;

            &::before {
                position: absolute;
                bottom: -4px;
                right: 0;
                width: 7px;
                height: 6px;
                background-color: $GREEN;
                content: "";
            }
        }
    }

    .expand-enter-active, .expand-leave-active {
        transform-origin: top left;
    }

    .expand-enter-active {
        animation: bounce-in 0.4s;
    }

    .expand-leave-active {
        animation: bounce-in 0.4s reverse;
    }

    @keyframes bounce-in {
        0% {
            transform: scale(1, 0);
        }

        100% {
            transform: scale(1);
        }
    }
</style>
