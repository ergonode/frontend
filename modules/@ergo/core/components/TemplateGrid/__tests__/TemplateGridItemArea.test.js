/*
 * Copyright Â© Bold Brand Commerce Sp. z o.o. All rights reserved.
 * See LICENSE for license details.
 */
import { shallowMount } from '@vue/test-utils';
import TemplateGridItemArea from '../TemplateGridItemArea';

describe('TemplateGrid/TemplateGridItemArea', () => {
    let wrapper;
    beforeEach(() => {
        wrapper = shallowMount(TemplateGridItemArea, {
            propsData: {
                item: {
                    id: 'test',
                    column: 1,
                },
                columns: 2,
            },
        });
    });

    it('Component is named well', () => {
        expect(typeof TemplateGridItemArea.name).toBe('string');
        expect(TemplateGridItemArea.name).toEqual('TemplateGridItemArea');
    });
    it('Component has main class', () => {
        expect(wrapper.contains('.grid__item-area')).toBeTruthy();
        expect(wrapper).toMatchSnapshot();
    });
    it('Check item data', () => {
        expect(wrapper.vm.item).toStrictEqual({
            column: 1,
            id: 'test',
        });
        expect(wrapper.contains('[item-id="test"]')).toBe(true);
    });
    it('Check item style with default props', () => {
        expect(wrapper.vm.gridItemStyles).toStrictEqual({
            gridColumn: '2 / 3',
            gridTemplateColumns: 'repeat(1, 1fr)',
            padding: '0px 0',
        });
    });
    it('Check item style with new props', () => {
        wrapper.setProps({
            item: {
                id: 'test',
                column: 3,
            },
            columns: 6,
            gridGap: 10,
        });
        expect(wrapper.vm.gridItemStyles).toStrictEqual({
            gridColumn: '4 / 7',
            gridTemplateColumns: 'repeat(3, 1fr)',
            padding: '10px 0',
        });
    });
});
