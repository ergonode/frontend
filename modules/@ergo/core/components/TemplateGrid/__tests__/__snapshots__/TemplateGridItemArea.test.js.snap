// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateGrid/TemplateGridItemArea Component has main class 1`] = `<div item-id="test" class="grid__item-area" style="grid-column: 2 / 3; padding: 0px 0px; grid-template-columns: repeat(1, 1fr);"> </div>`;
